icu4c/source/common/normalizer2impl.h

```diff
@@ -264,7 +264,9 @@ class U_COMMON_API Normalizer2Impl : public UObject {
     // The trie stores values for lead surrogate code *units*.
     // Surrogate code *points* are inert.
     uint16_t getNorm16(UChar32 c) const {
-        return U_IS_LEAD(c) ? INERT : UCPTRIE_FAST_GET(normTrie, UCPTRIE_16, c);
+        return U_IS_LEAD(c) ?
+            static_cast<uint16_t>(INERT) :
```

Comment:
If possible, I would like to keep this line unchanged.

Instead, with this internal enum and C++11, we should be able to use the syntax "enum : uint16_t { ..." -- see the anonymous enum on line 395 (unmodified file) with the comment "// Fixed norm16 values."

https://en.cppreference.com/w/cpp/language/enum

Reply:
Doesn't work. :(

If I change INERT's enum declaration to read `enum : uint16_t {` instead of `enum {` on line 395, and then remove the static_cast on line 267, I get the compiler warning again.

```
In file included from collationdata.h:24:0,
                 from utf8collationiterator.cpp:22:
../common/normalizer2impl.h: In member function ‘uint16_t icu_63::Normalizer2Impl::getNorm16(UChar32) const’:
../common/normalizer2impl.h:267:29: warning: enumeral and non-enumeral type in conditional expression [-Wextra]
         return U_IS_LEAD(c) ? INERT : UCPTRIE_FAST_GET(normTrie, UCPTRIE_16, c);
```

The header file seems to be popular, because this warning gets spammed several dozen times on a gcc build.